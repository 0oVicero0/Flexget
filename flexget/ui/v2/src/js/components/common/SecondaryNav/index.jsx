import React from 'react';
import PropTypes from 'prop-types';
import {
  SecondaryAppBar,
  SecondaryToolbar,
} from './styles';

const SecondaryNav = ({ children }) => (
  <SecondaryAppBar>
    <SecondaryToolbar>
      {children}
    </SecondaryToolbar>
  </SecondaryAppBar>
);

SecondaryNav.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SecondaryNav;
