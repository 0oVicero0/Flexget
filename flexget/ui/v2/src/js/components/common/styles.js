import styled from 'emotion/react';
import theme from 'theme';
import Paper from 'material-ui/Paper';

export const PaperWrapper = styled(Paper)`
  padding: 2.4rem;
  display: flex;
  flex: 1;
  height: 100%;
  box-sizing: border-box;
  flex-direction: column;
  ${theme.breakpoints.up('sm')} {
    padding-top: 0;
  }
`;
