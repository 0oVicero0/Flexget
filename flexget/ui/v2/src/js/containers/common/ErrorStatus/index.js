import { connect } from 'react-redux';
import StatusBar from 'components/common/StatusBar';
import { clearStatus } from 'actions/status';

export function mapStateToProps({ status }) {
  return {
    open: !!status.error,
    message: status.error,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(clearStatus()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatusBar);
