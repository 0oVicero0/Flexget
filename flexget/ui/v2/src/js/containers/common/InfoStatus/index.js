import { connect } from 'react-redux';
import StatusBar from 'components/common/StatusBar';
import { clearStatus } from 'actions/status';

export function mapStateToProps({ status }) {
  return {
    open: !!status.info,
    message: status.info,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(clearStatus()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatusBar);
