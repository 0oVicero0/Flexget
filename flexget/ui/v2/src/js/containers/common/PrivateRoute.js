import { connect } from 'react-redux';
import PrivateRoute from 'components/common/PrivateRoute';

function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export default connect(mapStateToProps)(PrivateRoute);
