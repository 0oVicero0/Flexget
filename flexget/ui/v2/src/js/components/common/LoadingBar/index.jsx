import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import { LinearProgress } from 'material-ui/Progress';

const styleSheet = theme => ({
  root: {
    position: 'relative',
    overflow: 'hidden',
    height: 5,
    backgroundColor: theme.palette.accent[100],
  },
  bar: {
    position: 'absolute',
    left: 0,
    bottom: 0,
    top: 0,
    transition: 'transform 0.2s linear',
    backgroundColor: theme.palette.accent[500],
  },
});


const LoadingBar = ({ loading, classes }) => {
  if (loading) {
    return (
      <LinearProgress
        mode="query"
        classes={{
          root: classes.root,
          bar: classes.bar,
        }}
      />
    );
  }
  return null;
};

LoadingBar.propTypes = {
  loading: PropTypes.bool,
  classes: PropTypes.object.isRequired,
};

LoadingBar.defaultProps = {
  loading: false,
};

export default withStyles(styleSheet)(LoadingBar);
