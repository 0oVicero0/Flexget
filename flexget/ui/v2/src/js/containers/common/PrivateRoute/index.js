import { connect } from 'react-redux';
import PrivateRoute from 'components/common/PrivateRoute';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export default connect(mapStateToProps)(PrivateRoute);
