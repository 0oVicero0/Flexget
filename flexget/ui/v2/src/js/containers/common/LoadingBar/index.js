import { connect } from 'react-redux';
import LoadingBar from 'components/common/LoadingBar';

export function mapStateToProps({ status }, { prefix = '@flexget' } = {}) {
  return {
    loading: !!(Object.keys(status.loading).find(type => type.startsWith(prefix))),
  };
}

export default connect(mapStateToProps)(LoadingBar);
