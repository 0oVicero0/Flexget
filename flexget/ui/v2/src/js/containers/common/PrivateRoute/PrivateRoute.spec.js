import { mapStateToProps } from 'containers/common/PrivateRoute';

describe('containers/common/PrivateRoute', () => {
  it('should return logged in if logged in', () => {
    expect(mapStateToProps({ auth: { loggedIn: true } })).toMatchSnapshot();
  });

  it('should return not logged in if logged out', () => {
    expect(mapStateToProps({ auth: { } })).toMatchSnapshot();
  });
});
