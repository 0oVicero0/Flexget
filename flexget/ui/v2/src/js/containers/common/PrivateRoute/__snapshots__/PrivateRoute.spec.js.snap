// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/common/PrivateRoute should return logged in if logged in 1`] = `
Object {
  "loggedIn": true,
}
`;

exports[`containers/common/PrivateRoute should return not logged in if logged out 1`] = `
Object {
  "loggedIn": false,
}
`;
